api/python/provisioner/commands/check.py

```diff
@@ -0,0 +1,392 @@
+        """
+        if dry_run:
+            return dict()
+
+        res = dict()
+
+        minion_id = local_minion_id()
+
+        cluster_path = KeyPath('cluster')
+        node_network_path = KeyPath(f'cluster/{minion_id}/network/data_nw')
+
+        pillar_keys = (PillarKey(cluster_path / 'type'),
+                       PillarKey(node_network_path / "public_ip_addr"),
+                       PillarKey(node_network_path / "pvt_ip_addr"),
+                       PillarKey(cluster_path / "cluster_ip"))
+
+        # maybe we need to call it from the all targets and analyze the output
+        # from each node
+        pillar = PillarResolver(cfg.LOCAL_MINION).get(pillar_keys)
+
+        pillar = pillar.get(local_minion_id())  # type: dict
+
+        for key in pillar_keys:
+            if not pillar[key] or pillar[key] is values.MISSED:
+                res[str(key.keypath)] = (f'{cfg.CheckVerdict.FAIL.value}: '
+                                         'value is not specified')
+            elif pillar[key] is None:
+                res[str(key.keypath)] = (f'{cfg.CheckVerdict.FAIL.value}: '
+                                         'value is unset')
+            else:
+                res[str(key.keypath)] = cfg.CheckVerdict.PASSED.value
+
+        return res
+
+    @staticmethod
+    def _communicability(*, args: str, targets: str = cfg.ALL_MINIONS,
+                         dry_run: bool = False) -> dict:
+        """
+        Check if nodes can communicate using `salt test.ping`
+
+        :param args: bmc accessibility check specific parameters and arguments
+        :param targets: target nodes where network checks will be executed
+        :param dry_run: Execute method without real check execution on
+                        target nodes
+        :return:
+        """
+        if dry_run:
+            return dict()
+
+        res = dict()
+
+        minion_id = local_minion_id()
+        try:
+            _res = function_run("test.ping", targets=minion_id)
+            # TODO: parse response
+            res[local_minion_id()] = (f"{cfg.CheckVerdict.PASSED.value}: "
+                                      f"{str(_res)}")
+        except SaltCmdResultError as e:
+            res[minion_id] = f"{cfg.CheckVerdict.FAIL.value}: {str(e)}"
+
+        return res
+
+    def _bmc_accessibility(self, *, args: str, targets: str = cfg.ALL_MINIONS,
+                           dry_run: bool = False) -> dict:
+        """
+        Check BMC accessibility
+
+        :param args: bmc accessibility check specific parameters and arguments
+        :param targets: target nodes where network checks will be executed
+        :param dry_run: Execute method without real check execution on
+                        target nodes
+        :return:
+        """
+        if dry_run:
+            return dict()
+
+        res = dict()
+        minion_id = local_minion_id()
+        try:
+            _res = function_run("cmd.run", targets=minion_id,
```

Comment:
https://github.com/Seagate/cortx-prvsnr/blob/9e5677a67a14480acc3f7690f33e003fee3a346f/api/python/provisioner/salt.py#L948

cmd_run has been improved and can now provide you all necessary features, you may use it directly

Reply:
Done